app/routes/route_utilities.py

```diff
@@ -0,0 +1,40 @@
+from flask import request, abort, make_response
+from ..db import db
+
+def validate_model(cls, id):
+    try: 
+        id = int(id)
+    except: 
+        abort(make_response({"details": "Invalid data"}, 400))
+    
+    query = db.select(cls).where(cls.id == id)
+    model = db.session.scalar(query)
+
+    if not model:
+        abort(make_response({"details": "Not found"}, 404))
```

Comment:
Ideally, we'd include the provided id in the response since we have that information. 

```suggestion
    if not model:
        abort(make_response({"details": f"{cls.__name__} {model_id} not found"}))
```